runners/flink/src/test/java/org/apache/beam/runners/flink/translation/wrappers/streaming/io/UnboundedSourceWrapperTest.java

```diff
@@ -707,4 +780,77 @@ public StreamStatus getStreamStatus() {
       return currentStreamStatus;
     }
   }
+
+  private static class BoundedIntegerSource extends OffsetBasedSource<Integer> {
```

Comment:
Hm,`CountingSource.upTo()` is deprecated. Otherwise it might be possible to reuse it here. Just a note

Reply:
There is also `CountingSource.createUnboundedFrom` which is package-private, but it looks like all the public methods are deprecated.

Reply:
I think it is deprecated for the sake of `GenerateSequence` which works on `PTransform` level. It uses `CountingSource` internally. Maybe it would make sense to be able to use also `source` level from this. But maybe there are not so many sources like this in tests

Reply:
Looking at all the other test which use `CountingSource.upTo()` I think it is probably ok to depend on this method, instead of essentially duplicating code. I've removed the source.